librz/core/cmd_api.c

```diff
@@ -1089,6 +1089,87 @@ static char *get_help(RzCmd *cmd, RzCmdDesc *cd, RzCmdParsedArgs *args, bool use
 	return NULL;
 }
 
+
+static bool child_help_json(RzCmd *cmd, RzCmdDesc *cd, PJ* j) {
+	pj_k(j, cd->name);
+	pj_o(j);
+	pj_ks(j, "cmd", cd->name);
+	if (cd->help->args_str) {
+		pj_ks(j, "args", cd->help->args_str);
+	} else {
+		RzStrBuf *sb = rz_strbuf_new(NULL);
+		fill_args(sb, cd);
+		char* args = rz_strbuf_drain(sb);
+		pj_ks(j, "args", args);
+		free (args);
+	}
+	pj_ks(j, "desc", cd->help->summary ? cd->help->summary : "");
+	pj_end(j);
+
+	return true;
+}
+
+static bool argv_modes_get_help_json(RzCmd *cmd, RzCmdDesc *cd, PJ *j) {
```

Comment:
Probably `argv_modes` and `argv` cmddesc printing can be refactored a lot to be almost equal.